includes/Core/Modules/Module.php

```diff
@@ -433,17 +433,25 @@ public function is_connected() {
 	 * @since 1.0.0
 	 *
 	 * @param Data_Request $data Data request object.
-	 *
 	 * @return mixed Data on success, or WP_Error on failure.
+	 *
+	 * phpcs:disable Squiz.Commenting.FunctionCommentThrowTag.Missing
 	 */
 	final protected function execute_data_request( Data_Request $data ) {
-		$client     = $this->get_client();
-		$orig_defer = $client->shouldDefer();
-		$client->setDefer( true );
+		$datapoint_services = $this->get_datapoint_services();
+
+		// We only need to initialize the client if this datapoint relies on a service.
+		if ( ! empty( $datapoint_services[ $data->datapoint ] ) ) {
```

Comment:
This conditional could be extracted to a more expressive variable here (perhaps `$request_requires_client`). This would make for more explicit conditionals in subsequent code, rather than simply checking on whether or not `$client` is set.